src/signatures/var_gen.rs

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_bls12_381::BlsScalar;
+use dusk_bytes::{DeserializableSlice, Error as BytesError, Serializable};
+use dusk_jubjub::{JubJubAffine, JubJubExtended, JubJubScalar};
+use dusk_poseidon::sponge::truncated::hash;
+
+use crate::PublicKeyVarGen;
+
+#[cfg(feature = "rkyv-impl")]
+use rkyv::{Archive, Deserialize, Serialize};
+
+#[cfg(feature = "zk")]
+use dusk_plonk::prelude::{Composer, Witness, WitnessPoint};
+
+/// An Schnorr SignatureVarGen, produced by signing a message with a
+/// [`SecretKeyVarGen`].
+///
+/// The `SignatureVarGen` struct encapsulates variables of the Schnorr scheme.
+///
+/// # Feature
+///
+/// Only available with the "var_generator" feature enabled.
+///
+/// ## Fields
+///
+/// - `u`: A [`JubJubScalar`] scalar representing the Schnorr signature.
+/// - `R`: A [`JubJubExtended`] point produced as part of the Schnorr signature.
+///
+/// ## Example
+///
+/// ```
+/// use dusk_bls12_381::BlsScalar;
+/// use jubjub_schnorr::{PublicKeyVarGen, SecretKeyVarGen, SignatureVarGen};
+/// use rand::rngs::StdRng;
+/// use rand::SeedableRng;
+/// use ff::Field;
+///
+/// let mut rng = StdRng::seed_from_u64(1234u64);
+///
+/// let sk = SecretKeyVarGen::random(&mut rng);
+/// let message = BlsScalar::random(&mut rng);
+/// let pk = PublicKeyVarGen::from(&sk);
+///
+/// // Sign the message
+/// let signature = sk.sign(&mut rng, message);
+///
+/// // Verify the signature
+/// assert!(pk.verify(&signature, message).is_ok());
+/// ```
+///
+/// [`SecretKeyVarGen`]: [`crate::SecretKeyVarGen`]
+#[allow(non_snake_case)]
+#[derive(Default, PartialEq, Clone, Copy, Debug)]
+#[cfg_attr(
+    feature = "rkyv-impl",
+    derive(Archive, Deserialize, Serialize),
+    archive_attr(derive(bytecheck::CheckBytes))
+)]
+pub struct SignatureVarGen {
+    u: JubJubScalar,
+    R: JubJubExtended,
+}
+
+impl SignatureVarGen {
+    /// Exposes the `u` scalar of the Schnorr SignatureVarGen.
+    pub fn u(&self) -> &JubJubScalar {
+        &self.u
+    }
+
+    /// Exposes the `R` point of the Schnorr SignatureVarGen.
+    #[allow(non_snake_case)]
+    pub fn R(&self) -> &JubJubExtended {
+        &self.R
+    }
+
+    /// Creates a new single key [`SignatureVarGen`] with the given parameters
+    #[allow(non_snake_case)]
+    pub(crate) fn new(u: JubJubScalar, R: JubJubExtended) -> Self {
+        Self { u, R }
+    }
+
+    /// Appends the single key as a witness to the circuit composed by the
+    /// [`Composer`].
+    ///
+    /// # Feature
+    ///
+    /// Only available with the "zk" feature enabled.
+    ///
+    /// ## Parameters
+    ///
+    /// - `composer`: Mutable reference to the Plonk `Composer`.
+    ///
+    /// ## Returns
+    ///
+    /// Returns a tuple `(Witness, WitnessPoint)` containing converted `u` and
+    /// `R` fields.
+    #[cfg(feature = "zk")]
+    pub fn append(&self, composer: &mut Composer) -> (Witness, WitnessPoint) {
```

Comment:
Not really sure if we should remove this. As said in the TODO that follows, the signature could be public. Even when we use it everywhere as a private value, we could need to make it public in other contexts. I would leave it to the circuit writer to choose all elements to be public or private, unless there is a very strong security reason to choose one option over the other.